pkg/kafka/auth/scram.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"crypto/sha512"
+	"fmt"
+	"hash"
+
+	"github.com/Shopify/sarama"
+	scrampkg "github.com/xdg/scram"
+)
+
+// ScramConfig describes the configuration properties required for the SCRAM handshake
+// between the collector and the pipeline
+type ScramConfig struct {
+	UserName  string `mapstructure:"username"`
+	Password  string `mapstructure:"password"`
+	Algorithm string `mapstructure:"algorithm"`
+}
+
+// SetSCRAMConfiguration ...
+func setSCRAMConfiguration(config *ScramConfig, saramaConfig *sarama.Config) error {
+	var fn func() sarama.SCRAMClient
+
+	var mechanism sarama.SASLMechanism
+
+	switch config.Algorithm {
```

Comment:
Matter of taste I guess, but I think the version from this PR more readable: https://github.com/open-telemetry/opentelemetry-collector/pull/2322